File: easy/p13-romanToInt/p13-romanToInt.cpp

#include <iostream>
#include <map>
#include <string>

using namespace std;

class Solution {
public:
    std::map<string, int> Roman;
    int romanToInt(string s) {
        return 0;
    }
};

int main() {
    Solution test;
    cout << test.romanToInt("III") << endl;     //3
    cout << test.romanToInt("IV") << endl;      //4
    cout << test.romanToInt("IX") << endl;      //9
    cout << test.romanToInt("LVIII") << endl;   //58
    cout << test.romanToInt("MCMXCIV") << endl; //1994
    cout << test.romanToInt("XLIX") << endl;    //49
    cout << test.romanToInt("CMXCIX") << endl;  //999
}

File: easy/p13-romanToInt/p13-romanToInt.py
class Solution:
    def __init__(self):
        self.m = {}
        self.m["IV"] = 4
        self.m["IX"] = 9
        self.m["XL"] = 40
        self.m["XC"] = 90
        self.m["CD"] = 400
        self.m["CM"] = 900
        self.m["I"] = 1
        self.m["V"] = 5
        self.m["X"] = 10
        self.m["L"] = 50
        self.m["C"] = 100
        self.m["D"] = 500
        self.m["M"] = 1000
    # def romanToInt(self, s: str) -> int:
    def romanToInt(self, s):
        num = 0
        while True:
            if len(s) == 0:
                break
            if len(s) >= 2:
                if s[0:2] in self.m.keys():
                    num += self.m[s[0:2]]
                    s = s[2:]
                else:
                    num += self.m[s[0]]
                    s = s[1:]
            else:
                num += self.m[s[0]]
                s = s[1:]
        return num

#IVXLCDM -1 5 10 50 100 500 1000
print(Solution().romanToInt("IV"))
print(Solution().romanToInt("MDCLXVI"))
print(Solution().romanToInt("DC"))
print(Solution().romanToInt("CD"))

